pkg/auth/oauth/external/handler.go

```diff
@@ -145,7 +145,7 @@ func DefaultState() State {
 func (defaultState) Generate(w http.ResponseWriter, req *http.Request) (string, error) {
 	state := url.Values{
 		"csrf": {"..."}, // TODO: get csrf
-		"then": {req.URL.String()},
+		"then": {url.QueryEscape(req.URL.String())},
```

Comment:
From what I can see, state is getting encoded correctly on our side. This is working with both Google and Github as external OAuth providers (queries wrapped for readability):

1. Console redirects to OpenShift OAuth /authorize endpoint:
>	GET https://localhost:8443/oauth/authorize?
>		client_id=openshift-web-console&
>		response_type=token&
>		state=%2F&
>		redirect_uri=https%3A%2F%2Flocalhost%3A8444%2Foauth

2. OpenShift OAuth /authorize endpoint redirects to Google:
>	< Location: https://accounts.google.com/o/oauth2/auth?
>		access_type=offline&
>		client_id=...&
>		include_granted_scopes=true&
>		redirect_uri=https%3A%2F%2Flocalhost%3A8443%2Foauth2callback%2Fgoogle&
>		response_type=code&
>		scope=profile+email&
>		state=csrf%3D...%26then%3D%252Foauth%252Fauthorize%253Fclient_id%253Dopenshift-web-console%2526response_type%253Dtoken%2526state%253D%25252F%2526redirect_uri%253Dhttps%25253A%25252F%25252Flocalhost%25253A8444%25252Foauth

3. Google does stuff

4. Google redirects back to OpenShift OAuth callback:
>	< Location: https://localhost:8443/oauth2callback/google?
>		state=csrf%3D...%26then%3D%252Foauth%252Fauthorize%253Fclient_id%253Dopenshift-web-console%2526response_type%253Dtoken%2526state%253D%25252F%2526redirect_uri%253Dhttps%25253A%25252F%25252Flocalhost%25253A8444%25252Foauth&
>		code=...&
>		scope=https://www.googleapis.com/auth/userinfo.profile+https://www.googleapis.com/auth/userinfo.email

5. OpenShift OAuth callback validates the code, parses the state, and redirects back to OpenShift /authorize:
>	< Location: /oauth/authorize?
>		client_id=openshift-web-console&
>		response_type=token&
>		state=%2F&
>		redirect_uri=https%3A%2F%2Flocalhost%3A8444%2Foauth